packages/workbox-build/src/entry-points/options/base-options.js

```diff
@@ -0,0 +1,56 @@
+const joi = require('joi');
```

Comment:
I'm offloading all parameter validation to `joi`, and running, it whenever one of the `entry-point/` modules are used, upfront, before any of the `lib/` modules are called.

As mentioned, the current approach of cascading/composed options is a little chunky, relying on class inheritance, so expect that to change in a follow-up PR. Custom error messages for specific validation failures are also to come in a follow-up.

If you have any comments about the overall approach to validation, usage of `joi`, etc. then that's fair game in this PR!